# Projeto: Análise Estratégica da Superstore | SCTEC
## Análise Exploratória de Dados (EDA)

### 1. Objetivo do Projeto
Este projeto consiste na análise exploratória de dados (EDA) da "Superstore", utilizando o dataset disponibilizado no [Kaggle](https://www.kaggle.com/datasets/vivek468/superstore-dataset-final). 

Alinhado à crescente demanda por uma cultura *data-driven* no polo tecnológico da Grande Florianópolis, este estudo vai além da exploração técnica. O objetivo central é **transformar dados brutos em insights acionáveis**, permitindo a identificação de gargalos logísticos e oportunidades reais de otimização de rentabilidade. Através de uma visão analítica sobre métricas de *Supply Chain* e margens por segmento, buscaremos fundamentar a tomada de decisão estratégica, demonstrando que decisões baseadas em dados são o motor da eficiência operacional e do crescimento sustentável.

---

### 2. Perguntas de Negócio
Para guiar nossa análise, investigaremos as seguintes hipóteses:

* **Eficiência Logística:** Qual o impacto do *lead time* (tempo de entrega) na performance operacional?
* **Rentabilidade:** Quais categorias de produtos apresentam margens de lucro abaixo do esperado ou negativas?
* **Sazonalidade:** Existe um padrão sazonal nas vendas que impacta a gestão de estoque e o fluxo de caixa?

---

### 3. Metodologia da Análise Exploratória (EDA)
A nossa exploração será estruturada da seguinte forma:

* **Auditoria de Qualidade:** Identificação de valores ausentes, dados duplicados e inconsistências estruturais.
* **Análise Univariada:** Investigação da distribuição estatística das variáveis principais (lucro, vendas e descontos).
* **Análise Temporal:** Avaliação do comportamento das vendas e envios ao longo do tempo (identificação de sazonalidade).
* **Análise de Correlação:** Investigação da sensibilidade do lucro em relação à estratégia de descontos aplicada.

## Importação de bibliotecas

In [8]:
import pandas as pd
from pathlib import Path
import numpy as np
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots

print("✓ Ambiente configurado e bibliotecas importadas.")

✓ Ambiente configurado e bibliotecas importadas.


## 1. Ingestão de Dados

In [9]:
# Definindo o caminho local usando pathlib (garante que funciona em Windows, Mac ou Linux)
# Aqui nos certificamos de que o dataset esteja em '/data'

file_path = Path('data/sample_superstore.csv')

try:
    # Carregamento com o encoding 'latin-1' (validado para este dataset)
    df = pd.read_csv(file_path, encoding='latin-1')
    print(f"✓ Dataset carregado com sucesso: {df.shape[0]} linhas, {df.shape[1]} colunas.")
except FileNotFoundError:
    print(f"❌ Erro: Arquivo não encontrado em {file_path}.")
    print("Dica: Verifique se a pasta 'data' existe e se o arquivo está lá.")

✓ Dataset carregado com sucesso: 9994 linhas, 21 colunas.


## 2. Auditoria de Integridade dos Dados (Sanity Check)

In [12]:
print("\n=== INFORMAÇÕES DO DATASET ===")
print(f"Forma: {df.shape}")
print(f"\nColunas:\n{df.dtypes}")
print(f"\nValores nulos por coluna:\n{df.isnull().sum()}")
print(f"\nPrimeiras 5 linhas:")
df.head()


=== INFORMAÇÕES DO DATASET ===
Forma: (9994, 21)

Colunas:
Row ID             int64
Order ID             str
Order Date           str
Ship Date            str
Ship Mode            str
Customer ID          str
Customer Name        str
Segment              str
Country              str
City                 str
State                str
Postal Code        int64
Region               str
Product ID           str
Category             str
Sub-Category         str
Product Name         str
Sales            float64
Quantity           int64
Discount         float64
Profit           float64
dtype: object

Valores nulos por coluna:
Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount     

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


O dataset foi carregado com **9.994 registros** e **zero valores nulos**, indicando uma alta qualidade estrutural inicial.

**Observações prática para as próximas etapas referente a tipagem de dados:**

As colunas `Order Date` e `Ship Date` estão como `str` (texto). A conversão para o formato `datetime` é o próximo passo necessário para habilitar a manipulação temporal (feature engineering) e viabilizar as análises de série temporal. Tal operação será realizada na seção 4.

# 3. Data Profiling (Estatísticas descritivas)

In [ ]:
print("\n=== ESTATÍSTICAS DESCRITIVAS ===")

# Primeiro obtemos médias, medianas, minímos e máximos.

print(df.describe())

# Verificar valores únicos em colunas categóricas, isso será útil para o design das futuras análises.

print(f"\nCategorias: {df['Category'].unique()}")
print(f"Segmentos: {df['Segment'].unique()}")
print(f"Regiões: {df['Region'].unique()}")


=== ESTATÍSTICAS DESCRITIVAS ===
            Row ID   Postal Code         Sales     Quantity     Discount  \
count  9994.000000   9994.000000   9994.000000  9994.000000  9994.000000   
mean   4997.500000  55190.379428    229.858001     3.789574     0.156203   
std    2885.163629  32063.693350    623.245101     2.225110     0.206452   
min       1.000000   1040.000000      0.444000     1.000000     0.000000   
25%    2499.250000  23223.000000     17.280000     2.000000     0.000000   
50%    4997.500000  56430.500000     54.490000     3.000000     0.200000   
75%    7495.750000  90008.000000    209.940000     5.000000     0.200000   
max    9994.000000  99301.000000  22638.480000    14.000000     0.800000   

            Profit  
count  9994.000000  
mean     28.656896  
std     234.260108  
min   -6599.978000  
25%       1.728750  
50%       8.666500  
75%      29.364000  
max    8399.976000  

Categorias: <StringArray>
['Furniture', 'Office Supplies', 'Technology']
Length: 3, dtype: 

A partir de estatísticas descritivas (média, mediana, desvio padrão), o diagnóstico numérico revelou dois comportamentos críticos que guiarão nossas próximas análises:

- **Assimetria (Sales):** A discrepância entre a média ($229) e a mediana ($54) confirma uma distribuição de "cauda longa". A maioria das vendas é de baixo valor, com poucos pedidos outliers elevando a média.
- **Volatilidade (Profit):** O lucro apresenta instabilidade severa. O desvio padrão ($234) é quase 10x superior à média ($28), indicando a presença de operações com prejuízos profundos que precisam ser isoladas.

# 4. Pré-processamento de Dados e Engenharia de Features

In [ ]:
# Pré processamento
# Converter datas para datetime
df['Order Date'] = pd.to_datetime(df['Order Date'], format='%m/%d/%Y')
df['Ship Date'] = pd.to_datetime(df['Ship Date'], format='%m/%d/%Y')
print("✓ Datas convertidas para datetime")

# Engenharia de features
# Criar colunas de tempo
df['Ano'] = df['Order Date'].dt.year
df['Mes'] = df['Order Date'].dt.month
df['Trimestre'] = df['Order Date'].dt.quarter
df['Data_Semana'] = (df['Ship Date'] - df['Order Date']).dt.days
print("✓ Colunas de tempo criadas (Ano, Mes, Trimestre, Data_Semana)")

# Verificar valores negativos em lucro (são válidos, empresas com prejuízo)
lucros_negativos = (df['Profit'] < 0).sum()
print(f"✓ Linhas com lucro negativo: {lucros_negativos} (mantidas, são dados válidos)")

# Verificar descontos acima de 100% (inválido)
descontos_invalidos = (df['Discount'] > 1).sum()
if descontos_invalidos > 0:
    print(f"⚠ Descontos > 100%: {descontos_invalidos} (podem ser removidos)")
else:
    print("✓ Todos os descontos estão no intervalo [0, 1]")

# Remover duplicatas completas (se houver)
duplicatas = df.duplicated().sum()
if duplicatas > 0:
    df = df.drop_duplicates()
    print(f"✓ {duplicatas} duplicatas removidas")
else:
    print("✓ Nenhuma duplicata encontrada")

print(f"\n✓ Dataset final: {df.shape[0]} linhas após limpeza")


=== LIMPEZA DE DADOS ===
✓ Datas convertidas para datetime
✓ Colunas de tempo criadas (Ano, Mes, Trimestre, Data_Semana)
✓ Linhas com lucro negativo: 1871 (mantidas, são dados válidos)
✓ Todos os descontos estão no intervalo [0, 1]
✓ Nenhuma duplicata encontrada

✓ Dataset final: 9994 linhas após limpeza


O pipeline de pré processamento de dados e engenharia de features foi executado com sucesso. O dataset encontra-se estruturado e validado para a fase de exploração. Esta etapa consolidou a preparação do dataset, garantindo que os dados estejam prontos para análises estatísticas e temporais robustas.

- **Integridade:** Tipos de dados convertidos (datetime) e ausência de registros duplicados confirmada.
- **Enriquecimento:** Criação de colunas temporais (`Ano`, `Mes`, `Trimestre`, `Data_Semana`) para viabilizar as análises temporais.
- **Validação de Negócio:** Confirmada a consistência das métricas financeiras. As 1.871 ocorrências de lucro negativo foram validadas como registros operacionais reais e mantidas para análise.
Além disso, validamos o intervalo de descontos (todos entre 0 e 1). Nenhum outlier ou valor inválido detectado.
